lms/resources/lti_launch.py

```diff
@@ -42,18 +42,21 @@ def get_or_create_course(self):
         )
 
         legacy_course = course_service.get_or_create(authority_provided_id)
+        new_legacy_course = legacy_course in self._request.db.new
```

Comment:
We can check the state down below, accessing the service again in line 47 does an implicit flush marking this as not new. 

Capturing the state here and checking it later.

Reply:
If this is important I'd definitely put a comment here. Not a chance in a million I'd spot that if I came to refactor.

Reply:
If this is important I'd definitely put a comment here. Not a chance in a million I'd spot that if I came to refactor.

Reply:
Github is being weird again. Not sure if this is coming through.

If we have an object for `legacy_course` then in generally I expect `new_legacy_course` to also be an object rather than a boolean.

I think a nice pattern for this is just to prefix booleans with `is_` to the extent it makes gramatical sense (it doesn't always). So we'd have:

* `legacy_course` - object
* `is_new_legacy_course` - bool

Reply:
Github is being weird again. Not sure if this is coming through.

If we have an object for `legacy_course` then in generally I expect `new_legacy_course` to also be an object rather than a boolean.

I think a nice pattern for this is just to prefix booleans with `is_` to the extent it makes gramatical sense (it doesn't always). So we'd have:

* `legacy_course` - object
* `is_new_legacy_course` - bool